subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/RetryConditions.groovy

```diff
@@ -129,6 +132,19 @@ class RetryConditions {
             return cleanProjectDir(specification)
         }
 
+        // known problem with Gradle versions < 3.5
+        // See https://github.com/gradle/gradle-private/issues/744
+        if (targetDistVersion < GradleVersion.version('3.5') && daemonsFixture != null && getRootCauseMessage(failure) == 'Build cancelled.') {
+            for (daemon in daemonsFixture.daemons) {
+                if (daemon.log.contains('Could not receive message from client.')
+                    && daemon.log.contains('java.lang.NullPointerException')
+                    && daemon.log.contains('org.gradle.launcher.daemon.server.exec.LogToClient$AsynchronousLogDispatcher$1.onOutput(LogToClient.java:80)')) {
```

Comment:
There is a chance that this is differs (slightly) in different Gradle versions. I think we can just remove this whole line to make sure we retry for every occurrence of the NPE. Since this check does not run for the current Gradle version, there is no danger in missing a new NPE by that.

Reply:
I find `NullPointerException` to broad. How about I check only for `LogToClient`? That class seems to have been there for ages.

Reply:
Sure. That's also fine.